web/src/components/MoreOptionsDropdown.tsx

```diff
@@ -0,0 +1,155 @@
+            )}
+          >
+            {isEstimated && (
+              <div className="w-full rounded-t-2xl bg-warning/10 p-3 text-xs font-semibold text-warning dark:bg-warning-dark/10 dark:text-warning-dark">
+                <p className="text-xs">{preliminaryData}</p>
+              </div>
+            )}
+            <div className="p-3">
+              <DropdownMenu.Label className="flex flex-col">
+                <div className="align-items flex justify-between">
+                  <h2 className="self-start text-sm">Share</h2>
+                  <DefaultCloseButton onClose={onDismiss} />
+                </div>
+                <TimeDisplay className="whitespace-nowrap text-xs text-neutral-600" />
+              </DropdownMenu.Label>
+              <DropdownMenu.Separator className="my-3 h-px bg-neutral-200" />
+              <DropdownMenu.Group
+                className={twMerge(
+                  'flex flex-col space-y-4 [&>div>p]:text-xs [&>div>p]:font-semibold [&>div]:flex [&>div]:items-center [&>div]:gap-2',
+                  'space-y-4 [&>a>div>p]:text-xs [&>a>div>p]:font-semibold [&>a>div]:flex [&>a>div]:gap-2'
+                )}
+              >
+                <DropdownMenu.Item onSelect={copyShareUrl}>
+                  <Copy size={DEFAULT_ICON_SIZE} />
+                  <p>Copy link to graph</p>
+                </DropdownMenu.Item>
+                {hasMobileUserAgent && (
+                  <DropdownMenu.Item onSelect={onShare}>
+                    <ShareIcon size={DEFAULT_ICON_SIZE} />
+                    <p>Share via </p>
+                  </DropdownMenu.Item>
+                )}
+                {!hasMobileUserAgent && (
+                  <>
+                    <a
+                      href={`https://twitter.com/intent/tweet?text=${encodeURI(
+                        summary
+                      )}&url=${shareUrl}&hashtags=electricitymaps`}
+                    >
+                      <DropdownMenu.Item>
+                        <FaXTwitter size={DEFAULT_ICON_SIZE} />
+                        <p>Share on X (Twitter)</p>
+                      </DropdownMenu.Item>
+                    </a>
+                    <a
+                      href={`https://facebook.com/sharer/sharer.php?u=${shareUrl}&quote=${encodeURI(
+                        summary
+                      )}`}
+                    >
+                      <DropdownMenu.Item>
+                        <FaFacebook size={DEFAULT_ICON_SIZE} />
+                        <p>Share on Facebook</p>
+                      </DropdownMenu.Item>
+                    </a>
+                    <a
+                      href={`https://www.linkedin.com/shareArticle?mini=true&url=${shareUrl}`}
+                    >
+                      <DropdownMenu.Item>
+                        <FaLinkedin size={DEFAULT_ICON_SIZE} />
+                        <p>Share on LinkedIn</p>
+                      </DropdownMenu.Item>
+                    </a>
```

Comment:
If it's not possible with the current one let's skip it for now and then we can look into it later.

I've been meaning to clean up the button component a bit anyway so it's easier to use in situations like this.

Reply:
I think that you're right that it would probably be ideal to have a single source of truth for social media sharing. I'd need to spend a little time looking into it. Maybe another follow up PR? 😓

Reply:
A follow up PR sounds great!